### PR TITLE
Added a Contribution Guide

### LINKED ISSUE #245: Add a Contribution Guide
Many Kotlin libraries like [KotlinDL](https://github.com/Kotlin/kotlindl/blob/master/CONTRIBUTING.md) or [kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/master/CONTRIBUTING.md) has pretty clear guides on how to contribute to the library

I suggest adding a clear guide and refer to this guide in the future.

### PR DESCRIPTION
**Pull request by zaleslaw**
Closes #245

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      excluding minor documentation changes, the addition of unit tests, and fixing typos.
+* If you make any code changes:
+    * Follow the [Kotlin Coding Conventions](https://kotlinlang.org/docs/reference/coding-conventions.html).
+    * [Build the project](#building) to make sure it all works and passes the tests.
+* If you fix a bug:
+    * Write the test the reproduces the bug.
+    * Fixes without tests are accepted only in exceptional circumstances if it can be shown that writing the
+      corresponding test is too hard or otherwise impractical.
+* If you introduce any new public APIs:
+    * All new APIs must come with documentation and tests.
```
Comment (koperagen) [inline CONTRIBUTING.md:42]: We have a little guide about contributions to documentation 
https://github.com/Kotlin/dataframe/blob/master/docs/contributions.md
Now apparently not many people see it

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    * Comment on the existing issue if you want to work on it.
+    * Wait till it is assigned to you by [maintainers](#contacting-maintainers).
+    * Ensure that the issue not only describes a problem, but also describes a solution that had received a positive feedback. Propose a solution if there isn't any.
+* If you are going to submit your first PR in this project:
+    * Find tickets with the label ["good first issue"](https://github.com/Kotlin/dataframe/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+no%3Aassignee)
+      which are not assigned to somebody.
+    * Learn the [`examples`](https://github.com/Kotlin/dataframe/tree/master/examples) module, submit new interesting example or improve documentation for one of them.
+* If you are ready to participate in library design and in new experiments, find tickets with the label
+  ["research"](https://github.com/Kotlin/dataframe/issues?q=is%3Aissue+is%3Aopen+label%3Aresearch)
+  or join to our [discussions](https://github.com/Kotlin/dataframe/discussions).
```
Comment (koperagen) [inline CONTRIBUTING.md:77]: I'm not sure about it, do people really use this functionality? Existing practise is discussions in issues.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  things up during development.
+
+You can import this project into IDEA, but you have to delegate build actions
+to Gradle (in Preferences -> Build, Execution, Deployment -> Build Tools -> Gradle -> Runner)
+
+## Contacting maintainers
+
+* If something cannot be done, not convenient, or does not work &mdash; submit an [issue](#submitting-issues).
+* To attract attention to the problem or raised question or a new comment, mention one of us on the Github: @koperagen @Jolanrensen @zaleslaw @ileasile
+* "How to do something" questions &mdash; [StackOverflow](https://stackoverflow.com).
```
Comment (koperagen) [inline CONTRIBUTING.md:98]: Personally, i never checked SO. There are few questions about our library. I see questions in Slack and issues more frequently

**Review by koperagen — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (zaleslaw) [reply CONTRIBUTING.md:77]: It depends on the community, but it's very good indexing and could be found via Google and some issues like our internal stuff could be moved to the discussions too

Comment (koperagen) [reply CONTRIBUTING.md:77]: Hmm, we can try to discuss some specific topic there :D I for instance found information from kotlinx.coroutines issues with Google as well. My concern is that having 2 different places where you can start a conversation as an external contributor raises the bar for indecisive people. But moving our discussions there might be a good idea

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      to gather community feedback.
+    * [Contact the maintainers](#contacting-maintainers) to coordinate any big piece of work in advance via submitting an issue.
+* If you fix documentation:
+    * If you plan extensive rewrites/additions to the docs, then please [contact the maintainers](#contacting-maintainers)
+      to coordinate the work in advance.
+    * Also, we have a special simple [guide](https://github.com/Kotlin/dataframe/blob/master/docs/contributions.md) how to contribute in the documentation.
+
+## PR workflow
+
+0. Contributor build the library locally and run all unit test via Gradle task `dataframe:test`
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:53]: The contributor builds the library locally and runs all unit tests via the Gradle task `dataframe:test`.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+   (see the ["Building"](#building) chapter).
+1. Contributor submits the PR if the local build is successful and tests are green.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:55]: *The contributor

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+2. Reviewer put his name to the "Reviewers" section of the proposed PR at the start of the review process.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:56]: *The reviewer puts their name...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+3. Reviewer leaves the comments or marks the PR with the word "LGTM."
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:57]: The reviewer leaves the comments or marks the PR with the abbreviation "LGTM" (Looks good to me).

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+4. Contributor answers the comments or fixes the proposed PR.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:58]: The

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+5. Reviewer marks the PR with the word "LGTM."
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:59]: The

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+6. Maintainer could suggest merging the master branch to the PR branch a few times due to changes in the `master` branch.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:60]: The

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+7. Maintainer runs TC builds (unit tests and examples as integration tests).
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:61]: The... (TeamCity)

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+8. The TC writes the result (passed or not passed) to the PR checks at the bottom of the proposed PR.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:62]: No "The" ;)

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+9. If it is possible, maintainers share the details of the failed build with the contributor.
+10. Maintainer merges the PR if all checks are successful and there is no conflict with the master branch.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:64]: The

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+## How to fix an existing issue
+
+* If you are going to work on the existing issue:
+    * Comment on the existing issue if you want to work on it.
+    * Wait till it is assigned to you by [maintainers](#contacting-maintainers).
+    * Ensure that the issue not only describes a problem, but also describes a solution that had received a positive feedback. Propose a solution if there isn't any.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:71]: had -> has

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+* If you are going to submit your first PR in this project:
+    * Find tickets with the label ["good first issue"](https://github.com/Kotlin/dataframe/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+no%3Aassignee)
+      which are not assigned to somebody.
+    * Learn the [`examples`](https://github.com/Kotlin/dataframe/tree/master/examples) module, submit new interesting example or improve documentation for one of them.
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:75]: ..module. Submit an interesting new example...

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+* Run `./gradlew build` to build. It also runs all the tests.
+* Run `./gradlew <module>:test` to test the module you are looking at to speed
+  things up during development.
+
+You can import this project into IDEA, but you have to delegate build actions
+to Gradle (in Preferences -> Build, Execution, Deployment -> Build Tools -> Gradle -> Runner)
+
+## Contacting maintainers
+
+* If something cannot be done, not convenient, or does not work &mdash; submit an [issue](#submitting-issues).
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:97]: *doesn't work conveniently

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+* To attract attention to the problem or raised question or a new comment, mention one of us on the Github: @koperagen @Jolanrensen @zaleslaw @ileasile
```
Comment (Jolanrensen) [inline CONTRIBUTING.md:98]: To attract attention to your problem, raised question, or new comment, mention one of us on Github:

**Review by Jolanrensen — CHANGES_REQUESTED** [review]
grammar fixes

**Comment by zaleslaw** [comment]
@Jolanrensen thanks a lot for your inspiration about the grammar, I have copied a lot from the KotlinDL guide and was sure that the grammar is ok, need to make a double check in the future before sending on the review

File: CONTRIBUTING.md (hunk 12 above)
Comment (Jolanrensen) [reply CONTRIBUTING.md:63]: @zaleslaw you forgot this one. "The TeamCity" is incorrect.

Comment (zaleslaw) [reply CONTRIBUTING.md:63]: Interesting, the Grammarly left this as a part of the text

Comment (Jolanrensen) [reply CONTRIBUTING.md:63]: Because it doesn't know what the abbreviation stands for :) It doesn't know it's a name

Comment (zaleslaw) [reply CONTRIBUTING.md:63]: Thanks, got it, will fix directly in the master

**Pull request merged**